File: rpr/insert.py
import re

def dialog():
  ret, orc_name, title, defex = RPR_GetUserFileNameForRead("/Users/j/Documents/PROJECTs/IDRA", "Path", "orc")
  if ret==1:
    return orc_name

def createtextitem(track, position, length, text):
    
  item = RPR_AddMediaItemToTrack(RPR_GetTrack(0, track))
  
  RPR_SetMediaItemInfo_Value(item, "D_POSITION", position)
  RPR_SetMediaItemInfo_Value(item, "D_LENGTH", length)
  
  retval, meditem, parname, instr, var = RPR_GetSetMediaItemInfo_String(item, "P_NOTES", text, 1)
  
  return item


def readfile(path):
  
  file = open(path, 'r')

  #grab instrument name
  instr_name = re.findall(r'"(.*?)"', file.read())
  instr_name = list(dict.fromkeys(instr_name))
  
  #RPR_ShowConsoleMsg(instr_name)
  
  file = open(path, 'r')
  
  for line in file:
    #i to schedulech
    #line = re.sub("^i", "schedulech\t", line)

    lineasarray = re.split('\t+', line)
    
    name = lineasarray[1].strip('\"')
    position = float(lineasarray[2])
    length = float(lineasarray[3])
    dyn = lineasarray[4]
    env = lineasarray[5]
    freq = lineasarray[6]
    
    score = '"' + name + '"\n' + dyn + '\n' + env + '\n' + freq
    
    #RPR_ShowConsoleMsg(score)
    
    for i, val in enumerate(instr_name):
      if name == val:
        createtextitem(i, position, length, score) 
        #RPR_ShowConsoleMsg(i)
        
  #closing files
  file.close() 
  

readfile(dialog())

File: rpr/*/insert.lua
function dialog()
  local ret, orc_name = reaper.GetUserFileNameForRead("/Users/j/Documents/PROJECTs/IDRA", "Path", "orc")
  if ret then
    return orc_name
  end
end

function createtextitem(track, position, length, text, color)
    
  local item = reaper.AddMediaItemToTrack(reaper.GetTrack(0, 0))
  
  reaper.SetMediaItemInfo_Value(item, "D_POSITION", position)
  reaper.SetMediaItemInfo_Value(item, "D_LENGTH", length)
  
  if text then
    reaper.ULT_SetMediaItemNote(item, text)
  end
  
  if color then
    reaper.SetMediaItemInfo_Value(item, "I_CUSTOMCOLOR", color)
  end
  
  return item

end

function readfile(path)
  local file = io.open(path, "rb") -- r read mode and b binary mode
  --if not file then return nil end
  for line in file:lines() do
    
    -- replace i to schedule
    line = line:gsub("^i", "schedulech")
    
    -- grab instrument name
    local allinstruments = {}
    local instrument = line:match('"(.+)"')
    
    -- put all instrument names inside an array
    table.insert(allinstruments, instrument)
    local flags = {}
    local allinstruments_res = {}
    
    for i in ipairs(allinstruments) do
       if not flags[allinstruments[i]] then
          table.insert(allinstruments_res, allinstruments[i])
          flags[allinstruments[i]] = true
       end
    end
    
    for i, v in ipairs(allinstruments_res) do
      reaper.ShowConsoleMsg(v .. "\n")
    end

    createtextitem(0, 0, 4, line)
  end
  file:close()
end

readfile(dialog())
